src/main/java/com/shaft/tools/io/PdfFileManager.java

```diff
@@ -0,0 +1,209 @@
+import org.apache.pdfbox.cos.COSDocument;
+import org.apache.pdfbox.io.RandomAccessBufferedFileInputStream;
+import org.apache.pdfbox.pdfparser.PDFParser;
+import org.apache.pdfbox.pdmodel.PDDocument;
+import org.apache.pdfbox.text.PDFTextStripper;
+import org.testng.Assert;
+
+import com.shaft.cli.FileActions;
+
+public class PdfFileManager {
+
+	private RandomAccessBufferedFileInputStream stream = null;
+	private PDFParser parser = null;
+	private COSDocument cosDoc = null;
+	private PDFTextStripper strip = null;
+	private File file;
+
+	/**
+	 * Validates the downloaded pdf document from the site under test
+	 * 
+	 * @param url                the path of the downloaded pdf file (usually will
+	 *                           be target/downloadedFiles/your-file.pdf), usually
+	 *                           will be in "target/downloadedFiles/" for the
+	 *                           project
+	 * @param milliSecondsToWait the time in milliseconds needed to wait on the
+	 *                           document to be downloaded successfully to the
+	 *                           target folder path
+	 * @param numberOfRetries    the retries that will be executed to check if the
+	 *                           file exist, will try every 5 seconds
+	 */
+	public PdfFileManager(String folderName, String fileName, int numberOfRetries) {
+
+		Boolean doesFileExist = FileActions.doesFileExist(folderName, fileName, numberOfRetries);
+
+		file = new File(FileActions.getAbsolutePath(folderName, fileName));
+
+		if (!doesFileExist) {
+			ReportManager.log("Couldn't find the provided file [" + file
+					+ "]. It might need to wait more to download or the path isn't correct");
+			Assert.fail("Couldn't find the provided file [" + file
+					+ "]. It might need to wait more to download or the path isn't correct");
+		}
+	}
+
+	public enum DeleteFileAfterValidationStatus {
+		TRUE, FALSE
+	}
+
+	/**
+	 * @param startPageNumber                 the starting page for the document to
+	 *                                        be validated
+	 * @param endPageNumber                   the ending page for the document to be
+	 *                                        validated
+	 * @param deleteFileAfterValidationStatus the status of deleting the file after
+	 *                                        get the document text
+	 * @return returns the pdf content in string so that can be validated
+	 */
+	public String readPDFContentFromDownloadedPDF(int startPageNumber, int endPageNumber,
+			DeleteFileAfterValidationStatus deleteFileAfterValidationStatus) {
+
+		stream = readFileInputStream(file);
+		parser = parseStreamDocument(stream);
+
+		cosDoc = getParsedDocument(parser);
+		String content = getPdfText(cosDoc, startPageNumber, endPageNumber);
+		closeStreamAndDeleteFile(file, stream, deleteFileAfterValidationStatus);
+
+		return content;
+	}
+
+	public String readPDFContentFromDownloadedPDF(DeleteFileAfterValidationStatus deleteFileAfterValidationStatus) {
+
```

Comment:
this method should have no parameters, and should call the above method and pass the default values to it. it shouldn't delete the file after validating it by default.